content/posts/2012-09-23-systemd-ionice.html

```diff
@@ -11,34 +11,63 @@
 </p>
 
 <pre>
-cp /lib/systemd/system/bacula-fd.service /etc/systemd/system/bacula-fd.service
+$ sudo systemctl enable bacula-fd.service
+Created symlink /etc/systemd/system/multi-user.target.wants/bacula-fd.service → /usr/lib/systemd/system/bacula-fd.service.
 </pre>
 
 <p>
-Then, open the file with an editor and change the following paragraph:
+Should this fail for you, you can also just link or copy the unit file by hand.
+</p>
+
+<p>
+Then, add an override for the service unit by doing the following:
 </p>
 
 <pre>
-[Service]
-Type=forking
-PIDFile=/var/run/bacula/bacula-fd.9102.pid
-StandardOutput=syslog
-ExecStart=/usr/sbin/bacula-fd -u root -g root -c /etc/bacula/bacula-fd.conf
+$ sudo systemctl edit bacula-fd.service
 </pre>
 
 <p>
-To look like this:
+This will open an editor to add additions to the service unit. depending on your
+systemd version you might not see the commented out options of the current unit.
```

Comment:
I think it would be clearer to not talk about/show the interactive editor view, but rather show the resulting drop-in configuration snippet (e.g. `/etc/systemd/system/bacula-fd.service.d/ionice.conf`)

Could you update accordingly please?

Reply:
Sure, I can see how it could be confusing. Also, I am also not sure which systemd version introduced showing the content of the unit as comments.

I find `systemctl cat <unit>` to be useful to show all files without explicitly knowing where to look for them. But the output could be a bit overwhelming:
```
$ sudo systemctl cat bacula-fd.service
# /lib/systemd/system/bacula-fd.service
# Systemd Bacula service file
#
# Copyright (C) 2000-2017 Kern Sibbald
# License: BSD 2-Clause; see file LICENSE-FOSS
#
# /lib/systemd/system/bacula-fd.service
#
# Description:
#    Used to start the bacula file daemon service (bacula-fd)
#
#    enable : systemctl enable bacula-fd
#    start : systemctl start bacula-fd
#
#

# from http://www.freedesktop.org/software/systemd/man/systemd.unit.html
[Unit]
Description=Bacula File Daemon service
Documentation=man:bacula-fd(8)
Requires=network.target
After=network.target
RequiresMountsFor=/var/lib/bacula /etc/bacula /usr/sbin

# from http://www.freedesktop.org/software/systemd/man/systemd.service.html
[Service]
Type=exec
User=root
Group=root
Environment="CONFIG=/etc/bacula/bacula-fd.conf"
EnvironmentFile=-/etc/default/bacula-fd
ExecStartPre=/usr/sbin/bacula-fd -t -c $CONFIG
ExecStart=/usr/sbin/bacula-fd -fP -c $CONFIG
ExecReload=/usr/sbin/bacula-fd -t -c $CONFIG
ExecReload=/bin/kill -HUP $MAINPID
SuccessExitStatus=15
Restart=on-failure
RestartSec=60

[Install]
WantedBy=multi-user.target

# /etc/systemd/system/bacula-fd.service.d/override.conf
[Service]
Nice=10
IOSchedulingClass=best-effort
IOSchedulingPriority=7
```

Reply:
I think we should mention `systemctl cat` for verification, but not show the full output. You can use “…” to indicate that you removed parts.